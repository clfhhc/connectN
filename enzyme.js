import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { setConfig } from 'next/config';
import config from './next.config';

// Make sure you can use "publicRuntimeConfig" within tests.
setConfig(config);
Enzyme.configure({ adapter: new Adapter() });
