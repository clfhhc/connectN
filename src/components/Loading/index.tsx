import React from 'react';
import css from 'styled-jsx/css';
import { rem } from '../../utils/styleUtils';

interface Props {
  boxSize?: number;
  width?: number;
  animationTime?: number;
  numBox?: number;
  boxColor?: string;
}

const Loading: React.FC<Props> = ({
  boxSize = 20,
  width = 180,
  animationTime = 1200,
  numBox = 3,
  boxColor = 'skyblue',
}) => {
  const animationDelayStyles = (position: number) => css.resolve`
    animation-delay: ${animationTime * (position / numBox)}ms;
  `;

  const boxStyles = Array.from(Array(numBox), (_, position) => animationDelayStyles(position));

  const containerStyles = css.resolve`
    box-sizing: border-box;
    margin-top: ${rem(40)};
    margin-bottom: ${rem(40)};
    margin-left: auto;
    margin-right: auto;
    display: flex;
    max-width: ${rem(width)};
    height: ${rem(boxSize * 2)};
    width: 100%;
    flex-flow: row;
    justify-content: space-evenly;
    align-items: center;

    @keyframes scale {
      0% {
        width: ${rem(boxSize)};
        height: ${rem(boxSize)};
        opacity: 0.5;
      }
      ${Math.floor(100 / numBox)}% {
        width: ${rem(boxSize * 2)};
        height: ${rem(boxSize * 2)};
        opacity: 1;
      }
      ${Math.floor(200 / numBox)}% {
        width: ${rem(boxSize)};
        height: ${rem(boxSize)};
        opacity: 0.5;
      }
      100% {
        width: ${rem(boxSize)};
        height: ${rem(boxSize)};
        opacity: 0.5;
      }
    }

    :global(${boxStyles.map(styles => `.${styles.className}`).join(', ')}) {
      width: ${rem(boxSize)};
      height: ${rem(boxSize)};
      opacity: 0.5;
      background-color: ${boxColor};
      animation-name: scale;
      animation-duration: ${animationTime}ms;
      animation-iteration-count: infinite;
    }
  `;

  return (
    <div className={containerStyles.className}>
      {boxStyles.map(styles => (
        <div key={styles.className} className={styles.className}>
          {styles.styles}
        </div>
      ))}
      {containerStyles.styles}
    </div>
  );
};

export default Loading;
