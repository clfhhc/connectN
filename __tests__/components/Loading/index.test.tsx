import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import { act } from 'react-dom/test-utils';
import toJson from 'enzyme-to-json';
import Loading from '../../../src/components/Loading';

describe('Loading', () => {
  let TestLoadingWrapper: ReactWrapper;

  beforeAll(() => {
    act(() => {
      TestLoadingWrapper = mount(<Loading />);
    });
  });

  afterAll(() => {
    TestLoadingWrapper.unmount();
  });

  it('should render without error', () => {
    expect(toJson(TestLoadingWrapper, { noKey: false, mode: 'deep' })).toMatchSnapshot();
  });
});
