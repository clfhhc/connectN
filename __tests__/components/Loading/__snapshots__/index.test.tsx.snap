// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render without error 1`] = `
<div
  className="jsx-551631717"
>
  <div
    className="jsx-660240691"
    key="jsx-660240691"
  />
  <div
    className="jsx-4081925047"
    key="jsx-4081925047"
  />
  <div
    className="jsx-4206995259"
    key="jsx-4206995259"
  />
</div>
`;
